source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -213,10 +206,12 @@ void GeoipProvider::lookupInAsnDb(
     absl::flat_hash_map<std::string, std::string>& lookup_result) const {
   if (config_->isLookupEnabledForHeader(config_->asnHeader())) {
     int mmdb_error;
-    auto isp_db = getIspDb();
-    RELEASE_ASSERT(isp_db, "Maxmind asn database must be initialised for performing lookups");
+    auto isp_db_ptr = getIspDb();
+    RELEASE_ASSERT(isp_db_ptr, "Maxmind asn database must be initialised for performing lookups");
+    auto isp_db = isp_db_ptr.get();
     MMDB_lookup_result_s mmdb_lookup_result = MMDB_lookup_sockaddr(
-        isp_db.get(), reinterpret_cast<const sockaddr*>(remote_address->sockAddr()), &mmdb_error);
+        &(isp_db->get()), reinterpret_cast<const sockaddr*>(remote_address->sockAddr()),
```

Comment:
I'm a bit surprised by this change - `isp_db` used to be a `shared_ptr<MMDB_s>` and is now, I think, a `const MMDB_s&`.

So `isp_db.get()` used to be a pointer to `MMDB_s`; wouldn't that just be `&isp_db` now, not `&(isp_db->get())`?

Oh, I see, isp_db_ptr is a shared_ptr, so isp_db is a bare pointer.

Two things then - could we *not* `auto` for this get, because it's unclear, and it would be much more readable to not go through three layers of get (getIspDb -> shared_ptr::get -> MaxmindDb::get).

`const MMDB_s& isp_db = isp_db_ptr->get();` would clarify what's happening and would skip the most confusing layer of get.

Reply:
Arguably even better alternative, add
```
const MMDB_s* mmdb() const {return &db_;}
```
to `MaxmindDb`, then you don't need an intermediate variable at all any more, you can just use `isp_db_ptr->mmdb()` directly.